src/libopenrave/libopenrave.cpp

```diff
@@ -1310,62 +1311,62 @@ MultiControllerBasePtr RaveCreateMultiController(EnvironmentBasePtr env, const s
 
 ModuleBasePtr RaveCreateModule(EnvironmentBasePtr penv, const std::string& name)
 {
-    return RaveGlobal::instance()->GetDatabase()->CreateModule(penv, name);
+    return RaveInterfaceCast<ModuleBase>(RaveGlobal::instance()->GetDatabase()->Create(penv, PT_Module, name));
```

Comment:
@undisputed-seraphim I think GetDatabase() ensures that interfaces returned from plugins have GetInterfaceType() match the expected type (If not, that's a bug then). Perhaps you can put the asserts in the database, and just use `boost::static_pointer_cast` directly here.

Reply:
@rdiankov  It turns out that the checks in `RaveInterfaceCast` [returns a null shared_ptr if the types don't match](https://github.com/rdiankov/openrave/blob/master/include/openrave/openrave.h#L2601), and it looks like some of the tests in autotester has come to rely on this behaviour (using an assert causes those tests to fail). I shall simulate that behaviour to maintain compatibility.

Reply:
Upon closer examination, the reason for that is because it's not a fatal condition to look for an interface but it can't be found, and `Create` just returns a null (shared) pointer and `RaveInterfaceCast` just passes the null pointer along.